motion_profiles.py

```diff
@@ -0,0 +1,101 @@
+"""Provide motion profiles for smooth and efficient motion"""
+
+from typing import NamedTuple
+
+
+class MotionProfile(NamedTuple):
+    """Motion profile representing a specific desired move"""
+
+    acceleration: float
+    deceleration: float
+    acceleration_time: float
+    deceleration_time: float
+    max_speed: float
+    end_time: float
+
+    @staticmethod
+    def intersection(line1, line2):
+        """Find where a pair of lines (represented as tuples
+         of (slope, intercept)) intersect
+        """
+        # Same slopes means parallel or collinear
+        if line1[0] == line2[0]:
```

Comment:
Showing `line1[0]`looks _magical_. Perhaps we should assign these values to variables to be more descriptive. Even `x1` and `y1` imply a coordinate.